multi-arch-builders/coreos-s390x-rhcos-builder.bu

```diff
@@ -91,3 +102,54 @@ storage:
 
             echo "Importing tarball into volume"
             sudo -u builder -H /bin/bash -c "cd /var/home/builder; podman volume import secex-data /var/home/builder/${TARBALL}"
+
+    - path: /usr/local/bin/create-cex-config.sh
+      mode: 0755
+      user:
+        name: root
+      group:
+        name: root
+      contents:
+        inline: |
+            #!/bin/bash
+            set -e
+
+            sudo modprobe vfio_ap
+            UUID='68cd2d83-3eef-4e45-b22c-534f90b16cb9'
+            if  ! ls /sys/devices/vfio_ap/matrix | grep -q "${UUID}"; then
+                cards=$(ls /sys/bus/ap/devices | grep -E 'card[0-9]{2}')
+                if [ -z "${cards}" ]; then
+                    echo "cex device not found."
+                    exit 1
+                fi
+
+              # if more than one card picks the first available cca controller
+              for card in $cards; do
+                  if $(grep -q cca /sys/bus/ap/devices/${card}/uevent); then
+                      card=${card}
+                      card_domain=$(ls /sys/bus/ap/devices/$card/ | grep -E '[0-9a-f]{2}.[0-9a-f]{4}')
+                      break
+                  fi
+              done
+
+              printf "Freeing adapter and domain \n"
+              echo 0x0 | sudo tee /sys/bus/ap/apmask
+              echo 0x0 | sudo tee /sys/bus/ap/aqmask
+              echo ${UUID} | sudo tee /sys/devices/vfio_ap/matrix/mdev_supported_types/vfio_ap-passthrough/create
```

Comment:
This code already runs as root

Reply:
I thought it runs as non-root user. Will correct it.